virttest/utils_disk.py

```diff
@@ -361,58 +363,85 @@ def create_partition_table(session, did, labeltype, ostype):
         create_partition_table_linux(session, did, labeltype)
 
 
-def create_partition_linux(session, did, size, start,
-                           part_type=PARTITION_TYPE_PRIMARY, timeout=360):
+def create_partition_linux(session, did, size, start=None,
+                           part_type=PARTITION_TYPE_PRIMARY,
+                           timeout=360):
     """
     Create single partition on disk in linux guest.
+    Note: Not support the decimal unit.
 
     :param session: session object to guest.
     :param did: disk kname. e.g. sdb
-    :param size: partition size. e.g. 200M
+    :param size: partition size. e.g. 200M or SIZE_AVAILABLE
     :param start: partition beginning at start. e.g. 0M
     :param part_type: partition type, primary extended logical
     :param timeout: Timeout for cmd execution in seconds.
     """
-    size = utils_numeric.normalize_data_size(size, order_magnitude="M") + "M"
-    start = utils_numeric.normalize_data_size(start, order_magnitude="M") + "M"
-    end = str(float(start[:-1]) + float(size[:-1])) + size[-1]
-    partprobe_cmd = "partprobe /dev/%s" % did
-    mkpart_cmd = 'parted -s "%s" mkpart %s %s %s'
-    mkpart_cmd %= ("/dev/%s" % did, part_type, start, end)
-    session.cmd(mkpart_cmd)
-    session.cmd(partprobe_cmd, timeout=timeout)
 
+    def _get_disk_free_space(unit='s'):
+        """Get the info of the disk free space."""
+        pattern = re.compile(r'(?P<start>\S+)\s+(?P<end>\S+)\s+'
+                             r'(?P<size>\S+)\s+Free\s+Space')
+        available_cmd = 'parted -s "/dev/%s" unit %s print free' % (did, unit)
+        spaces = session.cmd(available_cmd)
+        return pattern.search(spaces.splitlines()[-2]).groupdict()
+
+    unit = 'MiB'
+    if start is None and size == SIZE_AVAILABLE:
```

Comment:
Is there a need to call `_get_disk_free_space` every time `start`, `end` or ` size` is needed?